server/aesdsocket.c

```diff
@@ -311,15 +331,51 @@ void *connection_handler(void *client_data)
         if (ret_status < 0)
             goto close_client;
 
-        file_fd = open(SOCK_DATA_FILE, O_RDWR);
-#if !USE_AESD_CHAR_DEVICE
-        pthread_mutex_lock(&file_lock);
-#endif
-        ret_status = write(file_fd, client_node->malloc_buffer, malloc_buffer_len);
-#if !USE_AESD_CHAR_DEVICE
-        pthread_mutex_unlock(&file_lock);
-#endif
-        close(file_fd);
+        // Check if received string contains command
+        if (!memcmp(client_node->malloc_buffer, "AESDCHAR_IOCSEEKTO:", 19))
+        {
+            // Get position of X which should be just after ':'
+            write_cmd = memchr(client_node->malloc_buffer, ':', malloc_buffer_len);
+            
+            if ((client_node->malloc_buffer - write_cmd) >= (malloc_buffer_len - 1))
+                goto close_client;
+
+            // Get position of Y which should be just after ','
+            write_offset = memchr(client_node->malloc_buffer, ',', malloc_buffer_len);
+
+            if ((client_node->malloc_buffer - write_offset) >= (malloc_buffer_len - 1))
+                goto close_client;
+
+            write_cmd++;
+
+            if(write_cmd == write_offset)
+                goto close_client;
+
+            /* Convert character array starting from 'write_cmd' address to 'write_offset' 
+            to string which ends with null character */
+            *write_offset = '\0';
+
+            seekto.write_cmd = atoi(write_cmd);
```

Comment:
@ajaykandagal what does this code assume about the number of digits in write_cmd and write_offset?